src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,281 @@
+
+            # A built in python module or a local file, which have no version, only the python/SasView version
+            if module_name in standard_lib or module_name in package_versions_dict:
+                continue
+            # A built in python module or a local file, which have no version, only the python/SasView version
+            if package_name in standard_lib or package_name in package_versions_dict:
+                continue
+
+            # Modules that require specific methods to get the version number
+            if "PyQt5" in module_name:
+                try:
+                    from PyQt5.QtCore import PYQT_VERSION_STR
+                except Exception as x:
+                    # Unable to access PyQt5
+                    err_version_dict[module_name] = f"Unknown: {x} when attempting to get PyQt5 version"
+                    pass
+                else:
+                    package_versions_dict["PyQt"] = PYQT_VERSION_STR
+                try:
+                    from PyQt5.QtCore import QT_VERSION_STR
+                except Exception as x:
+                    # Unable to access Qt
+                    err_version_dict[module_name] = f"Unknown: {x} when attempting to get Qt version"
+                    pass
+                else:
+                    package_versions_dict["Qt"] = QT_VERSION_STR
+                continue
+
+            # Import module
+            try:
+                package = __import__(package_name)
+            except Exception as x:
+                err_version_dict[package_name] = f"Unknown: {x} when attempting to import module"
+                continue
+
+            # Different attempts of retrieving the modules version
+            if hasattr(package, '__version__'):
+                # Module has __version__ attribute
+                try:
+                    package_versions_dict[package_name] = package.__version__
+                except Exception as x:
+                    # Unable to access module
+                    err_version_dict[package_name] = f"Unknown: {x} when attempting to access {package_name} " \
+                                                     f"version using .__version__"
+                    pass
+                continue
+
+            if hasattr(package, '.version'):
+                # Module has .version attribute
+                try:
+                    package_versions_dict[package_name] = package.version
+                except Exception as x:
+                    # Unable to access module
+                    err_version_dict[package_name] = f"Unknown: {x} when attempting to access {package_name} " \
+                                                     f"version using .version"
+                    pass
+
+                continue
+
+            # Unreliable, so last option
+            try:
+                package_versions_dict[package_name] = pkg_resources.get_distribution(package_name).version
+
+            except:
```

Comment:
Don't use bare `except` statements. These capture interrupt and shutdown signals. At a minimum use `except Exception:`.